### ISSUE TITLE
Server crashes if node fails during insert into/select due to unnecessary assertion

### ISSUE THREAD

**Issue by velioglu** [citusdata/citus#2050]
It was shown up during 7.3 release tests. You can reproduce it by following the steps below.

```
-- tt1 is distributed on id column
create table tt1(id int, value_1 varchar(20));
insert into tt1 select x,x::text from generate_series(1,1000000) as f(x);
select create_distributed_table('tt1','id');

-- tt2 is distributed on value_1 column and data is inserted to it from
-- tt1 using coordinator insert into/select.
create table tt2(id int, value_1 varchar(20));
select create_distributed_table('tt2','value_1');
insert into tt2 select * from tt1; 
```

If you fail one of the worker nodes during the last command, server crashes with,

```
WARNING:  connection not open
WARNING:  failed to abort 1PC transaction "" on localhost:9700
ERROR:  server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
```

**Comment by marcocitus** [comment]
Is this a segmentation fault or an assert failure?

**Comment by onderkalaci** [comment]
It looks like an assertion failure:
```
Program received signal SIGABRT, Aborted.
0x00007fff96084f06 in __pthread_kill ()
(gdb) bt
#0  0x00007fff96084f06 in __pthread_kill ()
#1  0x00007fff92f4a4ec in pthread_kill ()
#2  0x00007fff9d7746df in abort ()
#3  0x000000010f238910 in ExceptionalCondition (conditionName=0x10f989ef8 "!(!result)", errorType=0x10f97b80e "FailedAssertion", fileName=0x10f989b8e "transaction/remote_transaction.c", lineNumber=448) at assert.c:54
#4  0x000000010f93f3b5 in FinishRemoteTransactionAbort (connection=0x7ffda3842aa8) at transaction/remote_transaction.c:448
#5  0x000000010f940106 in CoordinatedRemoteTransactionsAbort () at transaction/remote_transaction.c:891
#6  0x000000010f940c42 in CoordinatedTransactionCallback (event=XACT_EVENT_ABORT, arg=0x0) at transaction/transaction_management.c:203
#7  0x000000010ed1bf26 in CallXactCallbacks (event=XACT_EVENT_ABORT) at xact.c:3329
#8  0x000000010ed17232 in AbortTransaction () at xact.c:2574
#9  0x000000010ed181d1 in AbortCurrentTransaction () at xact.c:3003
#10 0x000000010f08dfbb in PostgresMain (argc=1, argv=0x7ffda401e578, dbname=0x7ffda401e3d8 "postgres", username=0x7ffda401e3b0 "onderkalaci") at postgres.c:3869
#11 0x000000010efe9762 in BackendRun (port=0x7ffda2503050) at postmaster.c:4342
#12 0x000000010efe89ad in BackendStartup (port=0x7ffda2503050) at postmaster.c:4016
#13 0x000000010efe7a05 in ServerLoop () at postmaster.c:1721
#14 0x000000010efe5246 in PostmasterMain (argc=3, argv=0x7ffda2500670) at postmaster.c:1329
#15 0x000000010ef15a7f in main (argc=3, argv=0x7ffda2500670) at main.c:228
```

I think I forgot to relax the assertions on #2031

**Comment by marcocitus** [comment]
> I think I forgot to relax the assertions on #2031

I think it's a separate issue. I've actually seen this assert trigger while simply running the regression tests: https://github.com/citusdata/citus/issues/1966

**Comment by onderkalaci** [comment]
> I think it's a separate issue. I've actually seen this assert trigger while simply running the regression tests: #1966

But after #2031 we're going to crash any every failure during abort? So, the following also hits the same assert:

```
postgres=# update tt1 set value_1 = id + 1;
UPDATE 1000000

-- stop the worker
postgres=# ROLLBACK;
WARNING:  connection not open
WARNING:  failed to abort 1PC transaction "" on localhost:9701
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
!> 

```

**Comment by onderkalaci** [comment]
It looks like we should simply remove the assert since it seems to make no sense?

**Comment by marcocitus** [comment]
Yeh, I think we can remove the `Assert`, but we probably also need to clear the result, so maybe replace `GetRemoteCommandResult` with `ClearResults`.

**Comment by marcocitus** [comment]
unrelated: `failed to abort 1PC transaction \"%s\"` is kind of weird since it never has a name

**Comment by velioglu** [comment]
As @onderkalaci mentioned, failing a node does not crash the server if asserts are not enabled.
